libfwupdplugin/fu-udev-device.c

```diff
@@ -793,6 +914,7 @@ fu_udev_device_set_physical_id (FuUdevDevice *self, const gchar *subsystems, GEr
 			return FALSE;
 		}
 		physical_id = g_strdup_printf ("PCI_SLOT_NAME=%s", tmp);
+		priv->bus_id = g_strdup (tmp);
```

Comment:
Why does bus-id need to be subsystem specific? Isn't the kernel interface to bind and unbind identical? Shouldn't all devices have this?

Reply:
For the PCI bus you use `PCI_SLOT_NAME`, for USB you use the USB bus ID, e.g. `1-1:1.0`

Reply:
But FuUsbDevice doesn't inherit from FuUdevDevice.  For USB the udev device isn't used at all I thought

Reply:
Agree; did you expect to see bus_id set for USB devices too?

Reply:
OK I think i see what you're meaning now looking at more context of this code.  Looking on my system at another device, i2c-hid:
```
/sys/module/i2c_hid/drivers/i2c:i2c_hid$ ls
bind  i2c-CUST0001:00  module  uevent  unbind
```
I can see that is actually the ACPI ID:
```
/sys/bus/acpi/devices$ ls -l | grep CUST
lrwxrwxrwx 1 root root 0 Sep 21 09:42 CUST0000:00 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:7d/CUST0000:00
lrwxrwxrwx 1 root root 0 Sep 21 09:42 CUST0001:00 -> ../../../devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:7e/CUST0001:00
```

So it makes me ask is there a more generic value that fits in here from resolving symlinks?

If not, I guess if/when it's needed outside of PCI the ID's used for other types can be added too.

Reply:
> Agree; did you expect to see bus_id set for USB devices too?

I did for i2c and USB too, yeah.

Reply:
> But FuUsbDevice doesn't inherit from FuUdevDevice. For USB the udev device isn't used at all I thought

Ahh, yes, now I see what you mean. What we probably need is a `fu_device_unbind_driver` on the base class, and then vfuncs implemented for FuUsbDevice and FuUdevDevice.

Reply:
Yep 👍

Reply:
I think I've done this; I don't have any i2c-hid devices handy so if you've got any more input for `fu_udev_device_get_bind_id()` that would be great. Unfortunately it's all very subsystem specific as it's all kinda evolved rather than with a nice abstracted sysfs key we can use.